code/game/machinery/doors/airlock.dm

```diff
@@ -117,6 +118,27 @@
 		max_integrity = normal_integrity
 	if(damage_deflection == AIRLOCK_DAMAGE_DEFLECTION_N && security_level > AIRLOCK_SECURITY_METAL)
 		damage_deflection = AIRLOCK_DAMAGE_DEFLECTION_R
+	if(abandoned)
+		var/outcome = rand(1,100)
+		switch(outcome)
+			if(1 to 9)
+				var/turf/here = get_turf(src)
+				for(var/turf/closed/T in range(2, src))
+					here.ChangeTurf(T.type)
+					return INITIALIZE_HINT_QDEL
+				here.ChangeTurf(/turf/closed/wall)
+				return INITIALIZE_HINT_QDEL
+			if(9 to 11)
+				lights = FALSE
+				bolt()
+			if(12 to 15)
+				bolt()
+			if(16 to 23)
+				welded = TRUE
+				update_icon()
+			if(24 to 30)
+				panel_open = TRUE
+				update_icon()
```

Comment:
You do not need to call `update_icon()`, as this is done at the end of the proc.